usaspending_api/reporting/v2/views/agencies/overview.py

```diff
@@ -0,0 +1,164 @@
+from django.db.models import Subquery, OuterRef, DecimalField, Func, F, Q, IntegerField
+from rest_framework.response import Response
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from django.utils.functional import cached_property
+
+from usaspending_api.common.data_classes import Pagination
+from usaspending_api.common.helpers.fiscal_year_helpers import (
+    get_final_period_of_quarter,
+    calculate_last_completed_fiscal_quarter,
+)
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.common.validator import customize_pagination_with_sort_columns, TinyShield
+from usaspending_api.references.models import ToptierAgency, Agency
+from usaspending_api.reporting.models import ReportingAgencyOverview, ReportingAgencyTas, ReportingAgencyMissingTas
+from usaspending_api.submissions.models import SubmissionAttributes
+
+
+class AgenciesOverview(AgencyBase):
+    """Return list of all agencies and the overview of their spending data for a provided fiscal year and period"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/reporting/agencies/overview.md"
+
+    def get(self, request):
+        results = self.get_agency_overview()
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        results = results[self.pagination.lower_limit : self.pagination.upper_limit]
+        return Response(
+            {
+                "page_metadata": page_metadata,
+                "results": results[: self.pagination.limit],
+                "messages": self.standard_response_messages,
```

Comment:
What are "messages"? It isn't in the contract; is it a standard part of REST responses that I'm not aware of?

Reply:
It's part of the Agency base class we use to explain why there is not pre-2017 data if the user selects a fiscal year prior to 2017: 
```
"messages": [
        "Account data powering this endpoint were first collected in FY2017 Q2 under the DATA Act; as such, there are no data available for prior fiscal years."
    ]
```

Reply:
Gotcha, good to know, thanks!

Reply:
It would be good to add `messages` to the API contract. Overlooked when defining the new endpoints